tests/roles/autoscaling_adoption/tasks/main.yaml

```diff
@@ -23,7 +23,7 @@
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
-    oc get pod --selector=service=aodh -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
+    oc wait pod --for condition=Ready --selector=service=aodh
```

Comment:
AFAIU this will wait indefinitely? That would be undesirable for CI, we should have some kind of timeout. I think `oc wait` supports it so maybe we can include `--timeout=120s` in these commands, and then we should also get rid of the `retries / delay / until` lines and most likely also `register` line because i don't think we use the `<service>_running_result` vars anywhere later.

Reply:
recheck the job is fixed now

Reply:
As per my observation in the CI run, it does not wait indefinitely but retries based on the ansible values set. Foe example, we leverage the ansible retry mechanism in keystone [1].
But i agree that we can get rid of a lot of ansible specific details by including the timeout which seems to be a good suggestion. Will update the commits.

[1] TASK [keystone_adoption : wait for Keystone to start up] ***********************
FAILED - RETRYING: [localhost]: wait for Keystone to start up (60 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (59 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (58 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (57 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (56 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (55 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (54 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (53 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (52 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (51 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (50 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (49 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (48 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (47 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (46 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (45 retries left).
FAILED - RETRYING: [localhost]: wait for Keystone to start up (44 retries left).
changed: [localhost] => {"attempts": 18, "changed": true, "cmd": "set -euxo pipefail\n\n\noc wait pod --for condition=Ready --selector=service=keystone\n", "delta": "0:00:00.307303", "end": "2024-09-25 02:15:24.155841", "msg": "", "rc": 0, "start": "2024-09-25 02:15:23.848538", "stderr": "+ oc wait pod --for condition=Ready --selector=service=keystone", "stderr_lines": ["+ oc wait pod --for condition=Ready --selector=service=keystone"], "stdout": "pod/keystone-86bcfb94dd-lhb62 condition met", "stdout_lines": ["pod/keystone-86bcfb94dd-lhb62 condition met"]}

Reply:
Ah yes indeed it is the case, thank you, the patch LGTM then :).